### PR TITLE
Update postgresql_privs.py to include `ALL x IN SCHEMA` functionality for PostgreSQL versions 9.x +

### LINKED ISSUE #269: postgresql_priv: Granting privileges using ALL_IN_SCHEMA does not use ALL TABLES in SCHEMA
##### SUMMARY
When using the postgresql_priv module, selecting the ALL_IN_SCHEMA option for the `obj` argument does not produce the expected statement.

EXPECTED: 
`GRANT <privileges> ON ALL TABLES IN SCHEMA <schema> TO <role>;`

ACTUAL:
`GRANT <privileges> ON schema.table1, schema.table2, schema.table3, ... schema.tableN <schema> TO <role>;`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

postgresql_priv

##### ANSIBLE VERSION

```paste below
ansible [core 2.12.4]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.4 (main, Apr  7 2022, 03:26:33) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
Collection           Version
-------------------- -------
community.postgresql 2.1.4
```

##### CONFIGURATION

```paste below
""
```

##### OS / ENVIRONMENT

Debian bullseye

##### STEPS TO REPRODUCE

Run the following statemens in a postgresql database:
CREATE TABLE public.table1(s varchar);
CREATE TABLE public.table2(s varchar);
CREATE TABLE public.table3(s varchar);
CREATE TABLE public.table4(s varchar);
CREATE TABLE public.table5(s varchar);
CREATE ROLE test;

Then run the following ansible task: 

```yaml
- name: "Example"
  postgresql_privs:
      login_host: "{{ host }}"
      login_user: "{{ user }}"
      login_password: "{{ password }}"
      db: postgres
      role: test
      priv: "SELECT"
      type: table
      objs: ALL_IN_SCHEMA
      schema: public
      state: present
```

##### EXPECTED RESULTS

The executed query should use the ALL TABLES IN SCHEMA {schema} statement.

##### ACTUAL RESULTS

The module builds a list of all tables in the schema and constructs a query specifying each one individually. This creates extremely long logs for databases with many tables.

```paste below
root@aa4d13fa304b:~/ansible$ ansible-playbook -vv test.yaml
ansible-playbook [core 2.12.4]
  config file = /root/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.10.4 (main, Apr  7 2022, 03:26:33) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True
Using /root/ansible/ansible.cfg as config file
redirecting (type: modules) ansible.builtin.postgresql_privs to community.postgresql.postgresql_privs
redirecting (type: callback) ansible.builtin.unixy to community.general.unixy
redirecting (type: callback) ansible.builtin.unixy to community.general.unixy
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
Executing playbook test.yaml
1 plays in test.yaml

- test on hosts: localhost -
Gathering Facts...
  localhost ok
META: ran handlers
Validate configuration json...
redirecting (type: modules) ansible.builtin.postgresql_privs to community.postgresql.postgresql_privs
  localhost ok: {
    "changed": false,
    "queries": [
        "GRANT SELECT ON table \"public\".\"table1\",\"public\".\"table2\",\"public\".\"table3\",\"public\".\"table4\",\"public\".\"table5\" TO \"test\";"
    ]
}
META: ran handlers
META: ran handlers
```

### PR DESCRIPTION
**Pull request by jchancojr**
Alter postgresql_privs.py to include `ALL x IN SCHEMA` functionality for PostgreSQL versions 9.0.0 or greater. We still allow versions 8.4.22 to `GRANT` to all SCHEMA, but this is done by building a list and granting to the objects on that list individually. 

Fixes #269 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
postgresql_privs.py

### CODE REVIEW

File: plugins/modules/postgresql_privs.py [hunk 1]
```diff
@@ -1093,14 +1096,28 @@ def main():
...
+                if p.type == 'table':
+                    orig_objs = 'ALL TABLES IN SCHEMA'
+                elif p.type == 'sequence':
+                    orig_objs = 'ALL SEQUENCES IN SCHEMA'
+                elif p.type == 'function':
+                    orig_objs = 'ALL FUNCTIONS IN SCHEMA'
+                elif p.type == 'procedure':
+                    orig_objs = 'ALL PROCEDURES IN SCHEMA'
+            else:
+                orig_objs = None
```
Comment (Andersson007) [inline plugins/modules/postgresql_privs.py:1119]:
```suggestion
```

File: plugins/modules/postgresql_privs.py [hunk 2]
```diff
@@ -1093,14 +1096,28 @@ def main():
...
         # objs:
-        if p.type == 'table' and p.objs == 'ALL_IN_SCHEMA':
-            objs = conn.get_all_tables_in_schema(p.schema)
-        elif p.type == 'sequence' and p.objs == 'ALL_IN_SCHEMA':
-            objs = conn.get_all_sequences_in_schema(p.schema)
-        elif p.type == 'function' and p.objs == 'ALL_IN_SCHEMA':
-            objs = conn.get_all_functions_in_schema(p.schema)
-        elif p.type == 'procedure' and p.objs == 'ALL_IN_SCHEMA':
-            objs = conn.get_all_procedures_in_schema(p.schema)
+        if p.objs == 'ALL_IN_SCHEMA':
```
Comment (Andersson007) [inline plugins/modules/postgresql_privs.py:1099]:
```suggestion
        orig_objs = None
        if p.objs == 'ALL_IN_SCHEMA':
```
i thin it'll fix the integration test error `UnboundLocalError: local variable 'orig_objs' referenced before assignment`

**Review by Andersson007 — COMMENTED** [review]
@jchancojr thanks for the PR!
The main question for me is if we should consider postgres 8.* or not..
As it seems toooo old and we don't test against it anyway, throwing out the related code can reduce the patch's complexity significantly. What do you folks think?

**Comment by hunleyd** [comment]
> The main question for me is if we should consider postgres 8.* or not..
> As it seems toooo old and we don't test against it anyway, throwing out the related code can reduce the patch's complexity significantly. What do you folks think?

I struggled with this very thing when we were hacking this together. The code as it works now, supports any PG version you throw at it. If we drop support for < 9.x this changes from a 'bugfix' to 'major breaking change'. And that seems really weird since we don't (yet) have a 'supported PG versions' list. So we opted for this approach with the knowledge that we'll rip it out in the near future when we formalize the support matrix.

**Comment by jchancojr** [comment]
Thanks @Andersson007 

I agree it's worth determining if workarounds like this for EOL versions are a good idea.

**Review by hunleyd — APPROVED** [review]

**Comment by Andersson007** [comment]
> > The main question for me is if we should consider postgres 8.* or not..
> > As it seems toooo old and we don't test against it anyway, throwing out the related code can reduce the patch's complexity significantly. What do you folks think?
> 
> I struggled with this very thing when we were hacking this together. The code as it works now, supports any PG version you throw at it. If we drop support for < 9.x this changes from a 'bugfix' to 'major breaking change'. And that seems really weird since we don't (yet) have a 'supported PG versions' list. So we opted for this approach with the knowledge that we'll rip it out in the near future when we formalize the support matrix.

Sounds good to me!

**Pull request merged**